extensions/src/org/pentaho/platform/web/http/api/resources/services/UserRoleDaoService.java

```diff
@@ -134,11 +140,49 @@ public void deleteRoles( String roleNames ) throws SecurityException, Uncategori
     }
   }
 
+  public SystemRolesMap getRoleBindingStruct( String locale ) throws SecurityException {
+    if ( canAdminister() ) {
+      RoleBindingStruct roleBindingStruct = getRoleBindingDao().getRoleBindingStruct( locale );
+      SystemRolesMap systemRolesMap = new SystemRolesMap();
+      for ( Map.Entry<String, String> localalizeNameEntry : roleBindingStruct.logicalRoleNameMap.entrySet() ) {
```

Comment:
can ```getRoleBindingDao().getRoleBindingStruct( locale );``` ever return null? if so, you might want to guard against that given an invalid/unsupported locale causing an NPE